### ISSUE TITLE
Shift key step yields "around" tick marks for most in Range 0-30

### ISSUE THREAD

**Issue by zepumph** [phetsims/ratio-and-proportion#298]
From https://github.com/phetsims/ratio-and-proportion/issues/279#issuecomment-743453138

@terracoda said:
>This seems to only be an issue for tick mark 1.
>Subsequent steps seem to be on mark visually and in the description.
>![Screen Shot 2020-12-11 at 6 38 28 PM](https://user-images.githubusercontent.com/1902116/101959551-18d11200-3be0-11eb-9585-d3f04315fe0d.png)

Then I said:

>@terracoda and I discussed this further today, and we see that this is because of using a shift keyboard step of 1/10 of an original tick mark, even when the range changes to 1/30. In this case, every tick mark that isn't divisible by 3 will be "around" because of the underlying math, not a rounding error. Over to @terracoda.

>Tagging some previous discussion about step size for range 0-30 in #175 and #166.

The way to solve this would be to change the shift key step to support being "on" all tick marks in range 0-30. I was worried that this would be challenging because of the purpose that the range tick marks overlay serves. I think of it as a change in representation, like a reskin that shouldn't effect the underlying model. It is possible that changing the input characteristics based on the range gives too much space to differentiate between range modes, rather than treating them all as different ways to see the same underlying model.

@terracoda mentioned that visually the range is an overlay. Visually you can still easily "get" to all lines (in which there is no snapping but you can adjust, but with the keyboard you can't. To her this feels like a bug because you have different input abilities based on modality.  

Next steps for this issue seem to be to have @emily-phet and @terracoda to remedy this discrepancy.

**Comment by terracoda** [comment]
I will review related issues and post my thoughts in the morning.
My main issue is that I find it weird that when using the keyboard, I cannot land "on" - only the ones divisible by 3 - when range is set to 0 - 30.

**Comment by terracoda** [comment]
Ok accidentally posted in #279 ... now re-posting in correct issue:

Ok, after a reset and then explore only with the keyboard on the 0 to 30 grid, I am always on a tick mark or on a half tick mark.

When I start using the mouse to get visually aligned with a visual tick mark, I get "around".

When I start using the keyboard again, the grid snaps back to being on a tick mark or on a half tick mark.

I think the smaller grid value that we discussed yesterday has improved the situation.

In fact, when I use the mouse, I can more easily get "around" now on any grid. That might have been the case before - not sure since I didn't test with the mouse much. Now, whenever I switch back to using the keyboard, I seem to snap back onto "on" a tick mark or "on" a half tick mark.

I'm thinking things are ok now.

I'd like to see, in the description, where we transition from "around" a tick mark to "around" a half tick mark. Sometime this might seem confusing, but maybe I am trying to be too precises in an imprecise environment.

**Comment by zepumph** [comment]
>I think the smaller grid value that we discussed yesterday has improved the situation.


I'm not sure I understand what you mean. I don't think there were any changes to the sim that addressed this issue.


>but maybe I am trying to be too precises in an imprecise environment.

I agree that in general the description has been designed to leave out a lot of precision. We are now seeing what that means for some edge cases in the sim. In general I like the current behavior. We also have to think about pedagogical scaffolding, and how likely, by the time range 0-30 is selected, the user will have spent some time with the sim, and gotten used to how it sounds in a "cleaner" condition (0-10 range). 

How would you like to proceed with this issue.

**Comment by terracoda** [comment]
Hmm, you say you didn't make any changes to the grid yesterday?
Well, maybe I just explored more thoroughly today, and am more comfortable with the design.

That said, I was reviewing the other issues, and I see back in issue #198, **I chose** the simple option (more vague option) of describing between the tick marks.

I am now thinking that the slightly less vague option of describing the points between the tick marks would make things a little more discernible within the impreciseness of the described experience. 

It may seem unrelated to this issue. In my mind, however, I feel everything is related.

How easy/hard would it be to use the less vague option from #198, copied here for clarity:

**For >1 to 10**
Small & Large steps | Tick Marks checked | Numbered Tick Marks checked
-- | -- | --
=10 | {{on}} {{third}} tick mark | {{on}} tick mark {{3}}
8-9 | **{{almost on}}** {{third}} tick mark | **{{almost on}}** tick mark {{3}}
6-7 | {{around}} {{half-way past}} {{second}} tick mark |{{around}} tick mark {{2.5}}
=5 | {{half-way past}} {{second}} tick mark | {{on}} tick mark {{2.5}}
3-4 | {{**almost** half-way past}} {{second}} tick mark | **{{almost on}}** tick mark {{2.5}}
1-2 | {{around}} {{second}} tick mark | {{around}} tick mark {{2}}
= 1 | {{on}} {{second}} tick mark | {{on}} tick mark {{2}}

**For between 1 and zero**

Small & Large steps | Tick Marks checked | Numbered Tick Marks checked
-- | -- | --
=10 | {{on}} {{first}} tick mark | {{on}} tick mark {{1}}
8-9 | **{{almost on}}** {{first}} tick mark | **{{almost on}}** tick mark {{1}}
6-7 | {{around}} {{half-way to}} {{first}} tick mark | {{around}} tick mark {{0.5}}
=5 | {{half-way to}} {{first}} tick mark | {{on}} tick mark {{0.5}}
3-4 | **{{almost}}** {{half-way to}} {{first}} tick mark | **{{almost on}}** tick mark {{0.5}}
1-2 | {{near}} tick mark {{zero}} | {{near}} tick mark {{zero}}
= 1 | {{at tick mark {{zero}}}} | {{at tick mark {{zero}}}}

**Comment by zepumph** [comment]
>How easy/hard would it be to use the less vague option from #198, copied here for clarity:


I don't know how to answer this question. If you think it is needed for the description, then we'll add it. We have already discussed that this is a very complicated set of regions, and that it was messy to implement. It will still be messy, as the above spec has no less edge cases as what is currently implemented. I think that one of your strong suites as a design is incredible attention to detail. With that, I feel like you are the most qualified to recommend how to proceed, because you know more intimately the value of a change like this.

There are currently 16 open bug issues for this repo. I don't feel like it is my job to determine the design of the description. It is up to you to determine if this change is valuable enough to add it to the list for me to do to get this sim out.

Another option is to defer this change until the next publication. 

Let me know how I can help further.

**Comment by terracoda** [comment]
Thanks, that's very helpful. Thanks for convincing me to stick up for my design :-)
I'd like to proceed with the update. 

I do think it is valuable, and I am kicking myself that I went with the less vague version to begin with (sorry for that).

Please note that the range for each each is exactly the same as before (i.e., as currently implemented). The difference is that for each region before the exact tick mark or exact half-way point, we replace "around" with "almost" or "almost on". I don't think I have conceptually introduced any new edge cases. There are still 10 tiny regions with the same ranges. The 4 bolded regions in each of the tables are where the changes are. In reality, there are 2 wording changes.

I hope that means that there are no new edge cases in the code.

Basically, when a hand is not exactly on a tick mark or on a half tick mark, this change would provide non-visual learners the ability to know if the hand is just above or just below the nearest tick mark.

**Comment by terracoda** [comment]
@zepumph, **I have reconsidered** when we should implement this change to the the ranges for in between the tick marks.

There are a lot of open issues for code review and dev test. I think it is more important to get the step sizes right in #275 and resolve issues in #288 and #289 before we make any more changes to the descriptions.

While I now feel the less vague scale will be an improvement, I think this change can be part of the next release, as you suggested in  https://github.com/phetsims/ratio-and-proportion/issues/298#issuecomment-746635601.

I'll mark this issue as low priority.

**Comment by terracoda** [comment]
"priority 5 - deferred" might be a better label, not sure :-)

**Comment by terracoda** [comment]
Oh no, I am mistaken, while in general I think things seem better. I can still get a description that is incorrect. In the screen shot the left hand seems to be clearly on Tick mark 2 and the and are described as being in ratio (2:3), but the position of the left hand is described as "around tick mark 2", not "on tick mark 2". 

Could this is just a rounding error? Have I asked this before?

Also, I noticed that when moving the left hand up and down is lands in different places and still hits ratio.

Note when using shift key and W/S to move up and down:
Starting at visual 2 (described as around tick mark 2) and going down one and then back up to visual 2 (described as around tick mark 2).
![Screen Shot 2021-03-09 at 11 02 30 AM](https://user-images.githubusercontent.com/1902116/110486543-4f2b0a00-80c7-11eb-9b32-f11cf25f82db.png)

Then starting again in the same place at visual 2 (described as around tick mark 2) and going up one and then back down to visual 2 (described as around tick mark 2). Strangely, the left hand is not exactly in the same place, though the step is still in ratio and the position is described as around tick mark 2. This time it looks like the left hand is more off the tick mark, even though in both cases the it should be on the tick mark and described as being on the tick mark.

I am leaving this issue open as it is weird. 
Maybe we can review it again before publication.
![Screen Shot 2021-03-09 at 11 01 50 AM](https://user-images.githubusercontent.com/1902116/110486638-64079d80-80c7-11eb-85d1-215d327ded44.png)

**Comment by terracoda** [comment]
Last comments are a bug not related to this design issue. Will move to a different issue.

**Comment by zepumph** [comment]
https://github.com/phetsims/ratio-and-proportion/issues/298#issuecomment-793987903 is related to https://github.com/phetsims/ratio-and-proportion/issues/354, but I still think it should be in a new issue like @terracoda said.

**Comment by zepumph** [comment]
I committed the design in https://github.com/phetsims/ratio-and-proportion/issues/298#issuecomment-746607402 above. I also made extensive unit tests to ensure the expected behavior. 

The only difference was that I found no hyphens in the current `halfway` wording. Would you like consistent hyphens in all `halfway` usages, like `half-way`?

**Comment by zepumph** [comment]
Oops, I forgot to re-assign. Sorry @terracoda.

**Comment by terracoda** [comment]
"halfway" seems to be a legitimate spelling. Let's go with no hyphens on "halfway" throughout. Would that be ok. I agree consistent spelling is important.

**Comment by terracoda** [comment]
And apologies for being inconsistent on that.

**Comment by terracoda** [comment]
I think this must be a stale issue since the sim is now published. Just checked ally-view of published version and I see "halfway" spelling. Apologies for not closing this sooner.
